app/policies/kudo_policy.rb

```diff
@@ -0,0 +1,23 @@
+class KudoPolicy < ApplicationPolicy
+  def update?
+    giver? && less_then_5_minutes_past_creation?
+  end
+
+  def edit?
+    update?
+  end
+
+  def destroy?
+    giver? && less_then_5_minutes_past_creation?
+  end
+
+  def giver?
+    @record.giver == @user
+  end
+
+  private
+
+  def less_then_5_minutes_past_creation?
+    (Time.current - @record.created_at).floor < 60 * 5
```

Comment:
You can compare '@record.created_at' with `5.minutes.ago`
Result will be `@record.created_at > 5.minutes.ago`

Docs: 
- [Active Support Duration (5.minutes)](https://api.rubyonrails.org/classes/ActiveSupport/Duration.html )
- [Time methods ( . ago)](https://api.rubyonrails.org/classes/Time.html#method-i-ago)